pkg/graph/drive.resolvers.go

```diff
@@ -462,6 +631,39 @@ func (r *queryResolver) Buckets(ctx context.Context, input *model.BucketFilterIn
 		return nil, err
 	}
 
+	// if filter shared is true, we also need to get the buckets that are shared with the user
+	if input != nil && input.Shared != nil && *input.Shared {
+		var shares []*db.Share
+		err = r.DB.NewSelect().
+			Model(&shares).
+			Where("shared_with = ?", currentUser.ID).
+			Scan(ctx)
+
+		// Get the bucket ids
+		var bucketIDs []string
+		for _, share := range shares {
+			bucketIDs = append(bucketIDs, share.BucketID)
+		}
+		var sharedBuckets []*db.Bucket
+		err = r.DB.NewSelect().
+			Model(&sharedBuckets).
+			Where("shared = ?", *input.Shared).
+			Where("id IN (?)", bun.In(bucketIDs)).
+			Scan(ctx)
+		if err != nil {
+			if err != sql.ErrNoRows {
+				return nil, err
+			}
+		}
+
+		buckets = append(buckets, sharedBuckets...)
```

Comment:
I think we should do this beforehand and have it within `one`-sql query. More performant and perhaps more secure

Reply:
How is it possible to put it in one query, where in one you clearly fetch by user_id in bucket, which won't fetch any shared buckets?